tests/Doctrine/Tests/DBAL/Functional/Schema/MySqlSchemaManagerTest.php

```diff
@@ -216,6 +219,52 @@ public function testDoesNotPropagateDefaultValuesForUnsupportedColumnTypes(): vo
         self::assertFalse($onlineTable->getColumn('def_blob_null')->getNotnull());
     }
 
+    public function testAlterTableExplicitColumnPosition(): void
+    {
+        $tableName = 'test_column_position';
+        $table     = new Table($tableName);
+        $table->addColumn('id', 'integer');
+        $table->addColumn('a', 'integer');
+        $table->addColumn('b', 'integer');
+
+        $this->schemaManager->createTable($table);
+
+        self::assertSame(['id', 'a', 'b'], array_keys($this->schemaManager->listTableColumns($tableName)));
+
+        $table2 = new Table($tableName);
+        $table2->addColumn('id', 'float');
+        $table2->addColumn('new', 'float');
+        $table2->addColumn('b', 'float');
+        $table2->addColumn('a', 'float');
+
+        $comparator = new Comparator();
+        $this->schemaManager->alterTable($comparator->diffTable($table, $table2));
+
+        self::assertSame(['id', 'new', 'b', 'a'], array_keys($this->schemaManager->listTableColumns($tableName)));
+    }
+
+    public function testAlterTableExplicitColumnPositionColumnNotFoundException(): void
+    {
+        $tableName = 'test_column_position_exception';
+        $table     = new Table($tableName);
+        $table->addColumn('id', 'integer');
+
+        $this->schemaManager->createTable($table);
+
+        $table2 = new Table($tableName);
+        $table2->addColumn('id', 'float');
+        $table2WithoutAColumn = clone $table2;
+        $table2->addColumn('a', 'float');
+
+        $comparator    = new Comparator();
+        $diff          = $comparator->diffTable($table, $table2);
+        $diff->toTable = $table2WithoutAColumn;
```

Comment:
Once the diff is generated from the two tables, why should it be possible to change the referenced table via the API?

Reply:
this is test - to test exception https://github.com/doctrine/dbal/pull/4276/files#diff-548b552a74ea52eab68318520c213a142ba77527a4b965c4d3bd50eeba1afa9aR675

Reply:
This test was meant to be added to document the case when this exception can happen because IMO it shouldn't be possible in the first place (https://github.com/doctrine/dbal/pull/4276/files#r493065750). The test doesn't show a valid case. Which means that there is no justification for this API to throw an exception.

Reply:
When the diff object is built using Comparator::diff which will add the input, then this exception can not happen.

However, the `toTable` is public property - like `fromTable` already present - and if the table is replaced by the user with different table that has not all the column names set - this exception is legit and this test cover it.